src/pages/workspaces/workspace/Data.test.ts

```diff
@@ -28,31 +26,49 @@ jest.mock('src/libs/ajax', (): AjaxExports => {
   };
 });
 
+// When Data.js is broken apart and the WorkspaceData component is converted to TypeScript,
+// this type belongs there.
+interface WorkspaceDataProps {
+  namespace: string;
+  name: string;
+  workspace: WorkspaceWrapper;
+  refreshWorkspace: () => void;
+  storageDetails: StorageDetails;
+}
+type AjaxContract = ReturnType<typeof Ajax>;
+
 describe('WorkspaceData', () => {
-  // When Data.js is broken apart and the WorkspaceData component is converted to TypeScript,
-  // this type belongs there.
-  interface WorkspaceDataProps {
+  type SetupOptions = {
```

Comment:
@nawatts at your suggestion, I added these types in for `Setup`, but I have two questions:

* does it make sense to keep them local to the `describe`?  I thought other test cases might have their own version of `setup` and therefore these belonged here rather than at the module level
* Just curious about the use of terms `Options` vs. `Props` -- later on I have a type called `WorkspaceDataProps` which could also, in my mind, just be named `WorkspaceDataOptions` -- are there conventions on when one name is used vs. the other?

Reply:
> does it make sense to keep them local to the describe?

Yes, I think so. I think my comments about the scope for type declarations were tied to thinking of the common case where a module exports one thing and its test has one describe block.

> Just curious about the use of terms Options vs. Props

"Props" is a React term. https://react.dev/learn#sharing-data-between-components

So the convention is that a React function component takes one argument: a "props" object.

Vs "options" is a more generic term for functions in general.
https://www.codereadability.com/what-are-javascript-options-objects